src/c_stuff_sack.py

```diff
@@ -0,0 +1,545 @@
+  }
+
+  root_type = type_object.root_type
+
+  if root_type.name in type_mapping:
+    return type_mapping[root_type.name]
+
+  return root_type.name
+
+
+def c_field_name(field):
+  length_list = []
+  current_type = field.type
+  while isinstance(current_type, ss.Array):
+    length_list.append(current_type.length)
+    current_type = current_type.type
+
+  brackets = ''.join('[{}]'.format(x) for x in length_list)
+
+  return '{}{}'.format(field.name, brackets)
+
+
+def pack_function_name(obj):
+  return 'SsPack{}'.format(_snake_to_camel(obj.name))
+
+
+def unpack_function_name(obj):
+  return 'SsUnpack{}'.format(_snake_to_camel(obj.name))
+
+
+def declaration(type_object):
+  if type(type_object) is ss.Primitive:
+    return None
+
+  if isinstance(type_object, ss.Bitfield):
+    return bitfield_declaration(type_object)
+
+  if isinstance(type_object, ss.Enum):
+    return enum_declaration(type_object)
+
+  if isinstance(type_object, ss.Struct):
+    return struct_declaration(type_object)
+
+  raise TypeError('Unknown type: {}'.format(type(type_object)))
+
+
+def bitfield_c_type(bitfield):
+  return 'uint{}_t'.format(bitfield.bytes * 8)
+
+
+def bitfield_field_declaration(bitfield, field):
+  return '{} {} : {}'.format(bitfield_c_type(bitfield), field.name, field.bits)
+
+
+def bitfield_declaration(bitfield):
+  s = ''
+  s += 'typedef struct {\n'
+
+  for field in bitfield.fields:
+    s += _indent('{};\n'.format(bitfield_field_declaration(bitfield, field)))
+
+  s += '}} {};'.format(bitfield.name)
+
+  return s
+
+
+def enum_declaration(enum):
+  s = ''
+  s += 'typedef enum {\n'
+  s += _indent('k{}ForceSigned = -1,\n'.format(enum.name))
+
+  for value in enum.values:
+    s += _indent('k{}{} = {},\n'.format(enum.name, value.name, value.value))
+
+  s += _indent('kNum{} = {},\n'.format(enum.name, len(enum.values)))
+  s += '}} {};'.format(enum.name)
+
+  return s
+
+
+def struct_field_declaration(field):
+  return '{} {}'.format(c_type_name(field.type), c_field_name(field))
+
+
+def struct_declaration(struct):
+  s = ''
+  s += 'typedef struct {\n'
+
+  for field in struct.fields:
+    s += _indent('{};\n'.format(struct_field_declaration(field)))
+
+  s += '}} {};'.format(struct.name)
+
+  return s
+
+
+def pack(type_object):
+  if isinstance(type_object, ss.Bitfield):
+    return primitive_union_pack(type_object)
+
+  if type_object.name in ('float', 'double'):
+    return primitive_union_pack(type_object)
+
+  if isinstance(type_object, ss.Enum):
+    return primitive_pack(type_object)
+
+  if isinstance(type_object, ss.Message):
+    return message_pack(type_object)
+
+  if isinstance(type_object, ss.Struct):
+    return struct_pack(type_object)
+
+  if isinstance(type_object, ss.Primitive):
+    return primitive_pack(type_object)
+
+  raise TypeError('Unknown type: {}'.format(type(type_object)))
+
+
+def unpack(type_object):
+  if isinstance(type_object, ss.Bitfield):
+    return primitive_union_unpack(type_object)
```

Comment:
Can you add a comment why this is handled by union_unpack instead of primitive_unpack?